client.c

```diff
@@ -0,0 +1,99 @@
+#include <stdio.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <arpa/inet.h>
+
+#define B_SIZE 1024
+
+int main(int argc)
+{
+  /* local variables declaration */
+  int clientSocket, status, addr_size;
+  char buffer[B_SIZE], file_name[100];
+  char *name;
+  struct sockaddr_in serverAddr;
+  FILE *file_1;
+  
+  /* create TCP, stream client socket */
+  clientSocket = socket(PF_INET, SOCK_STREAM, 0);
+
+  if(clientSocket < 0)
+  {
+    printf("Error %d in creating client socket!\n", errno);
+    return 1;
+  }
+ 
+   /* initialise sockaddr and buffer to 0 */
+  memset(&serverAddr, 0, sizeof(serverAddr));
+  memset(buffer, 0, sizeof(buffer));
+
+  /* update sockaddr fields */
+  serverAddr.sin_family = AF_INET;
+  serverAddr.sin_port = htons(5000);  // how to transmit server port to client?  
+  serverAddr.sin_addr.s_addr = inet_addr("127.0.0.1");
+
+  /* get sockaddr size */
+  addr_size = sizeof serverAddr;
+
+  /* connect to server */
+  status = connect(clientSocket, (struct sockaddr *)&serverAddr, addr_size);
+
+  if(status < 0)
+  {
+    printf("Error %d in connecting to server socket!\n", errno);
```

Comment:
you cannot use the socket in the next instructions if it was not succesfully connected to server so return is needed